File: searching/linearSearch.js
const linearSearch = (arr = [], key) => {
  let index = -1;
  arr.forEach((ele, i) => {
    if (ele === key) index = i;
  });
  return index;
};

console.log(linearSearch([1, 2, 3, 4, 5, 6, 7, 8, 9], 10));
console.log(linearSearch([1, 2, 3, 4, 5, 6, 7, 8, 9], 5));

File: searching/linear_search.go
// Linear Search in Golang,

package main

import "fmt"

func main() {
	items := []int{1, 12, 13, 14, 17, 99, 100, 21, 33}
	fmt.Println(linearSearch(items, 12))
}

func linearSearch(array []int, key int) int {
	for index, value := range array {
		if key == value {
			return index
		}
	}
	return -1
}
